In [384]:
import pandas as pd
import plotly.express as px
import plotly.graph_objects as go
import seaborn as sns
import matplotlib.pyplot as plt


pd.set_option('display.max_columns', None)

In [296]:
data = pd.read_csv('./Speed+Dating+Data.csv', encoding='ISO-8859-1')

In [297]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8378 entries, 0 to 8377
Columns: 195 entries, iid to amb5_3
dtypes: float64(174), int64(13), object(8)
memory usage: 12.5+ MB


There are 195 columns in the dataframe !!!

In [298]:
data.isna().sum().sum()

432016

Quite a lot of NA (because some data depends on if there was a followup date or not) ! Remember to check if selected column has NA before proceeding

In [299]:
data['iid'].nunique()

551

551 persons

We are only looking for informations during first date (and especially what will lead to a followup date), so we need to understand which data are relevant only for date nb 1

For the attributes:

-- example of attractiveness, for Date 1 = attrx_1
  

Now we need to understand how attributes for first dates are structured : 



attr1_1 = 'What you look for in the opposite sex (Waves 6-9 = 1-10 scale, Waves 1-5; 10-21 = 100pt alloc)'

attr4_1 = 'What you think most of your fellow men/women look for in the opposite sex (Waves 6-9 =  1-10 scale, Waves 10-21 = 100pt alloc)'

attr2_1 = 'What do you think the opposite sex looks for in a date (Waves 6-9 = 1-10 scale, Waves 1-5; 10-21 = 100pt alloc)'

attr3_1 = 'What is your opinion on your own attributes? (1-10 scale)'

attr5_1 = 'How do you think others perceive you? (1-10 scale)'

Finally,

dec = decision of the person

dec_o = decision of the mate

## option: How many dates led to a follow_up ?

In [300]:
data['dec'].info()

<class 'pandas.core.series.Series'>
RangeIndex: 8378 entries, 0 to 8377
Series name: dec
Non-Null Count  Dtype
--------------  -----
8378 non-null   int64
dtypes: int64(1)
memory usage: 65.6 KB


In [301]:
followup = (data['dec'] == 1) & (data['dec_o'] == 1)
data_follow_up = data[followup]
len(data_follow_up)

1380

Only 1380 out of of 8378 dates led to a followup

In [302]:
followup_count = len(data_follow_up)
non_followup_count = len(data) - followup_count
pie_data = {'Dates': ['Followup', 'No Followup'], 'Count': [followup_count, non_followup_count]}
pie_df = pd.DataFrame(pie_data)

pie_follow = px.pie(pie_df, names='Dates', values='Count', title='How many first dates led to a follow up?', color='Dates', color_discrete_map={'Followup': '#ffaf3c', 'No Followup': '#3c5587'})
pie_follow.update_layout(
    width=400,
    height=400,
    margin=dict(l=20, r=20, t=50, b=20),  # Reduce the margins
    title=dict(font=dict(size=15), x=0.5),
    legend=dict(font=dict(size=10))
)
pie_follow.show()

In [303]:
data.head(3)

iid   id  gender  idg  condtn  wave  round  position  positin1  order  \
0    1  1.0       0    1       1     1     10         7       NaN      4   
1    1  1.0       0    1       1     1     10         7       NaN      3   
2    1  1.0       0    1       1     1     10         7       NaN     10   

   partner   pid  match  int_corr  samerace  age_o  race_o  pf_o_att  \
0        1  11.0      0      0.14         0   27.0     2.0      35.0   
1        2  12.0      0      0.54         0   22.0     2.0      60.0   
2        3  13.0      1      0.16         1   22.0     4.0      19.0   

   pf_o_sin  pf_o_int  pf_o_fun  pf_o_amb  pf_o_sha  dec_o  attr_o  sinc_o  \
0      20.0      20.0      20.0       0.0       5.0      0     6.0     8.0   
1       0.0       0.0      40.0       0.0       0.0      0     7.0     8.0   
2      18.0      19.0      18.0      14.0      12.0      1    10.0    10.0   

   intel_o  fun_o  amb_o  shar_o  like_o  prob_o  met_o   age field  field_cd  \
0      8.0    8.0    8.0     6.0     7.0     4.0    2.0  21.0   Law       1.0   
1     10.0    7.0    7.0     5.0     8.0     4.0    2.0  21.0   Law       1.0   
2     10.0   10.0   10.0    10.0    10.0    10.0    1.0  21.0   Law       1.0   

  undergra mn_sat tuition  race  imprace  imprelig     from zipcode  \
0      NaN    NaN     NaN   4.0      2.0       4.0  Chicago  60,521   
1      NaN    NaN     NaN   4.0      2.0       4.0  Chicago  60,521   
2      NaN    NaN     NaN   4.0      2.0       4.0  Chicago  60,521   

      income  goal  date  go_out  career  career_c  sports  tvsports  \
0  69,487.00   2.0   7.0     1.0  lawyer       NaN     9.0       2.0   
1  69,487.00   2.0   7.0     1.0  lawyer       NaN     9.0       2.0   
2  69,487.00   2.0   7.0     1.0  lawyer       NaN     9.0       2.0   

   exercise  dining  museums  art  hiking  gaming  clubbing  reading   tv  \
0       8.0     9.0      1.0  1.0     5.0     1.0       5.0      6.0  9.0   
1       8.0     9.0      1.0  1.0     5.0     1.0       5.0      6.0  9.0   
2       8.0     9.0      1.0  1.0     5.0     1.0       5.0      6.0  9.0   

   theater  movies  concerts  music  shopping  yoga  exphappy  expnum  \
0      1.0    10.0      10.0    9.0       8.0   1.0       3.0     2.0   
1      1.0    10.0      10.0    9.0       8.0   1.0       3.0     2.0   
2      1.0    10.0      10.0    9.0       8.0   1.0       3.0     2.0   

   attr1_1  sinc1_1  intel1_1  fun1_1  amb1_1  shar1_1  attr4_1  sinc4_1  \
0     15.0     20.0      20.0    15.0    15.0     15.0      NaN      NaN   
1     15.0     20.0      20.0    15.0    15.0     15.0      NaN      NaN   
2     15.0     20.0      20.0    15.0    15.0     15.0      NaN      NaN   

   intel4_1  fun4_1  amb4_1  shar4_1  attr2_1  sinc2_1  intel2_1  fun2_1  \
0       NaN     NaN     NaN      NaN     35.0     20.0      15.0    20.0   
1       NaN     NaN     NaN      NaN     35.0     20.0      15.0    20.0   
2       NaN     NaN     NaN      NaN     35.0     20.0      15.0    20.0   

   amb2_1  shar2_1  attr3_1  sinc3_1  fun3_1  intel3_1  amb3_1  attr5_1  \
0     5.0      5.0      6.0      8.0     8.0       8.0     7.0      NaN   
1     5.0      5.0      6.0      8.0     8.0       8.0     7.0      NaN   
2     5.0      5.0      6.0      8.0     8.0       8.0     7.0      NaN   

   sinc5_1  intel5_1  fun5_1  amb5_1  dec  attr  sinc  intel  fun  amb  shar  \
0      NaN       NaN     NaN     NaN    1   6.0   9.0    7.0  7.0  6.0   5.0   
1      NaN       NaN     NaN     NaN    1   7.0   8.0    7.0  8.0  5.0   6.0   
2      NaN       NaN     NaN     NaN    1   5.0   8.0    9.0  8.0  5.0   7.0   

   like  prob  met  match_es  attr1_s  sinc1_s  intel1_s  fun1_s  amb1_s  \
0   7.0   6.0  2.0       4.0      NaN      NaN       NaN     NaN     NaN   
1   7.0   5.0  1.0       4.0      NaN      NaN       NaN     NaN     NaN   
2   7.0   NaN  1.0       4.0      NaN      NaN       NaN     NaN     NaN   

   shar1_s  attr3_s  sinc3_s  intel3_s  fun3_s  amb3_s  sa

## Now on to the main questions : 

1) What are the least desirable attributes in a male partner? Does this differ for female partners?

2) How important do people think attractiveness is in potential mate selection vs. its real impact?

3) Are shared interests more important than a shared racial background?

4) Can people accurately predict their own perceived value in the dating market?


## 1) What are the least desirable attributes in a male partner? Does this differ for female partners?

To answer this question, we need to look at **attributes**1_1 per gender ! Be mindful that ratings are different per certaiens waves, so should be adjusted for. 

example : attr1_1 = 'What you look for in the opposite sex (Waves 6-9 = 1-10 scale, Waves 1-5; 10-21 = 100pt alloc)'

So should analyze this per gender for waves 6-9 & for waves(1-5 & 10-21)

male = 1, female = 0

In [304]:
data['gender'].value_counts()

gender
1    4194
0    4184
Name: count, dtype: int64

In [305]:
male_alloc = (data['gender']==1) & (data['wave'].between(1,5)) | (data['gender']==1) & (data['wave'].between(10,21))
male_scale = (data['gender']==1) & (data['wave'].between(6,9))
female_alloc = (data['gender']==0) & (data['wave'].between(1,5)) | (data['gender']==0) & (data['wave'].between(10,21))
female_scale = (data['gender']==0) & (data['wave'].between(6,9))

data_male_alloc = data[male_alloc]
data_male_scale = data[male_scale]
data_female_alloc = data[female_alloc]
data_female_scale = data[female_scale]

In [306]:
# Define columns of interest
attributes_partn_firstd = ['attr1_1', 'sinc1_1', 'intel1_1', 'fun1_1', 'amb1_1','shar1_1']

### For men rating women

In [307]:
data_male_alloc_trim = data_male_alloc[attributes_partn_firstd].copy() # Create a copy to ignore slice warning
data_male_alloc_trim.rename(columns={'attr1_1': 'Attractive', 'sinc1_1': 'Sincere', 'intel1_1': 'Intelligent', 'fun1_1': 'Fun', 'amb1_1': 'Ambitious','shar1_1': 'Commonalities'}, inplace=True)
mean = data_male_alloc_trim.mean()
median = data_male_alloc_trim.median()
data_male_alloc_trim_stat = pd.DataFrame({'Attributes': mean.index,'Mean': mean.values, 'Median': median.values})
data_male_alloc_trim_stat.sort_values('Mean', ascending=True, inplace=True)
data_male_alloc_trim_stat

Attributes       Mean  Median
4      Ambitious   7.498773     8.0
5  Commonalities  10.258607    10.0
1        Sincere  16.233127    16.0
3            Fun  17.657753    20.0
2    Intelligent  19.564615    20.0
0     Attractive  29.108895    25.0

In [308]:
data_male_scale_trim = data_male_scale[attributes_partn_firstd].copy() # Create a copy to ignore slice warning
data_male_scale_trim.rename(columns={'attr1_1': 'Attractive', 'sinc1_1': 'Sincere', 'intel1_1': 'Intelligent', 'fun1_1': 'Fun', 'amb1_1': 'Ambitious','shar1_1': 'Commonalities'}, inplace=True)
stat = data_male_scale_trim.mean()
median = data_male_scale_trim.median()
data_male_scale_trim_stat = pd.DataFrame({'Attributes': mean.index,'Mean': mean.values, 'Median': median.values})
data_male_scale_trim_stat.sort_values('Mean', ascending=True, inplace=True)
data_male_scale_trim_stat

Attributes       Mean  Median
4      Ambitious   7.498773   13.95
5  Commonalities  10.258607   15.09
1        Sincere  16.233127   17.78
3            Fun  17.657753   17.78
2    Intelligent  19.564615   19.44
0     Attractive  29.108895   17.24

In [309]:
web_male_alloc = go.Figure()

web_male_alloc.add_trace(go.Scatterpolar(
      r=data_male_alloc_trim_stat['Mean'],
      theta=data_male_alloc_trim_stat['Attributes'],
      fill='toself',
      name='Mean'
))
web_male_alloc.add_trace(go.Scatterpolar(
      r=data_male_alloc_trim_stat['Median'],
      theta=data_male_alloc_trim_stat['Attributes'],
      fill='toself',
      name='Median'
))

web_male_alloc.update_layout(
    title='How Points Were Distributed Across Desired Women Attributes by Men',
    polar=dict(
        radialaxis=dict(
            visible=True,
            range=[0, 35],
            tickfont=dict(size=13)
        ),
        angularaxis=dict(
            tickfont=dict(size=15)
        )
    ),
    showlegend=True,
    width=600,
    height=600,
    font=dict(size=12)
)

web_male_alloc.show()

In [310]:
px.histogram(data_male_scale_trim_stat, x='Attributes', y=['Mean', 'Median'], barmode='group', title='How desired attributes among Women were rated by Men')

### And now for women rating men

In [311]:
data_female_alloc_trim = data_female_alloc[attributes_partn_firstd].copy() # Create a copy to ignore slice warning
data_female_alloc_trim.rename(columns={'attr1_1': 'Attractive', 'sinc1_1': 'Sincere', 'intel1_1': 'Intelligent', 'fun1_1': 'Fun', 'amb1_1': 'Ambitious','shar1_1': 'Commonalities'}, inplace=True)
mean = data_female_alloc_trim.mean()
median = data_female_alloc_trim.median()
data_female_alloc_trim_stat = pd.DataFrame({'Attributes': mean.index,'Mean': mean.values, 'Median': median.values})
data_female_alloc_trim_stat.sort_values('Mean', ascending=True, inplace=True)
data_female_alloc_trim_stat

Attributes       Mean  Median
4      Ambitious  12.005635    11.0
5  Commonalities  12.264193    10.0
3            Fun  17.043556    17.0
1        Sincere  18.377089    20.0
0     Attractive  18.788883    16.0
2    Intelligent  21.568459    20.0

In [312]:
web_female_alloc = go.Figure()

web_female_alloc.add_trace(go.Scatterpolar(
      r=data_female_alloc_trim_stat['Mean'],
      theta=data_female_alloc_trim_stat['Attributes'],
      fill='toself',
      name='Mean'
))
web_female_alloc.add_trace(go.Scatterpolar(
      r=data_female_alloc_trim_stat['Median'],
      theta=data_female_alloc_trim_stat['Attributes'],
      fill='toself',
      name='Median'
))

web_female_alloc.update_layout(
    title='How Points Were Distributed Across Desired Men Attributes by Women',
    polar=dict(
        radialaxis=dict(
            visible=True,
            range=[0, 35],
            tickfont=dict(size=13)
        ),
        angularaxis=dict(
            tickfont=dict(size=15)
        )
    ),
    showlegend=True,
    width=600,
    height=600,
    font=dict(size=12)
)

web_female_alloc.show()

In [313]:
data_female_scale_trim = data_female_scale[attributes_partn_firstd].copy() # Create a copy to ignore slice warning
data_female_scale_trim.rename(columns={'attr1_1': 'Attractive', 'sinc1_1': 'Sincere', 'intel1_1': 'Intelligent', 'fun1_1': 'Fun', 'amb1_1': 'Ambitious','shar1_1': 'Commonalities'}, inplace=True)
mean = data_female_scale_trim.mean()
median = data_female_scale_trim.median()
data_female_scale_trim_stat = pd.DataFrame({'Attributes': mean.index,'Mean': mean.values, 'Median': median.values})
data_female_scale_trim_stat.sort_values('Mean', ascending=True, inplace=True)
data_female_scale_trim_stat

Attributes       Mean  Median
5  Commonalities  14.590644   14.89
0     Attractive  14.888956   15.38
4      Ambitious  16.372964   16.98
3            Fun  17.594987   18.00
1        Sincere  17.993930   18.00
2    Intelligent  18.559987   18.37

In [314]:
px.histogram(data_female_scale_trim_stat, x='Attributes', y=['Mean', 'Median'], barmode='group', title='How desired attributes among Men were rated by Women')

## 2) How important do people think attractiveness is in potential mate selection vs. its real impact?

attr_4_1 vs attr1_1 accross the two waves

attr1_1 = 'What you look for in the opposite sex (Waves 6-9 = 1-10 scale, Waves 1-5; 10-21 = 100pt alloc)'

attr4_1 = 'What you think most of your fellow men/women look for in the opposite sex (Waves 6-9 =  1-10 scale, Waves 10-21 = 100pt alloc)'

In [315]:
attr_pot_vs_real = ['attr1_1', 'attr4_1', 'wave', 'gender']
data_attr = data[attr_pot_vs_real]
data_attr.head(1)

attr1_1  attr4_1  wave  gender
0     15.0      NaN     1       0

In [316]:
(data_attr['attr4_1'].isna().sum())/(len(data_attr))*100

22.547147290522798

22 % Missing values in attr4_1 -> to drop !

In [317]:
data_attr = data_attr.dropna(subset=['attr4_1'])

In [318]:
data_attr['wave'].value_counts().sort_index(ascending=True)

wave
6      45
7     512
8     200
9     800
10    162
11    882
12    392
13    160
14    702
15    684
16     96
17    280
18     72
19    450
20     84
21    968
Name: count, dtype: int64

NA values corresponded only to waves 1 to 5

Split by waves to make values comparable

In [319]:
attr_alloc = (data_attr['wave'].between(1,5)) | (data_attr['wave'].between(10,21))
attr_scale = (data_attr['wave'].between(6,9))

data_attr_alloc = data_attr[attr_alloc]
data_attr_scale = data_attr[attr_scale]

In [320]:
data_attr_scale['wave'].value_counts().sort_index(ascending=True)

wave
6     45
7    512
8    200
9    800
Name: count, dtype: int64

In [321]:
len(data_attr_scale)

1557

In [322]:
len(data_attr_alloc)

4932

In [325]:
attr_stat_scale = {
    'Method': ['What people look for', 'What people look for', 'What people think others look for', 'What people think others look for'],
    'Statistic': ['Mean', 'Median', 'Mean', 'Median'],
    'Value': [data_attr_scale['attr1_1'].mean(), data_attr_scale['attr1_1'].median(), data_attr_scale['attr4_1'].mean(), data_attr_scale['attr4_1'].median()]
}

In [326]:
attr_stat_scale

{'Method': ['What people look for',
  'What people look for',
  'What people think others look for',
  'What people think others look for'],
 'Statistic': ['Mean', 'Median', 'Mean', 'Median'],
 'Value': [16.15830443159923, 16.0, 8.638407193320488, 9.0]}

In [327]:
px.bar(attr_stat_scale, x='Statistic', y='Value', color='Method', barmode='group', title='Comparison of real (blue) vs projected (red) impact of attractiveness, regardless of sex (0-20 rates)')

In [328]:
attr_stat_alloc = {
    'Method': ['What people look for', 'What people look for', 'What people think others look for', 'What people think others look for'],
    'Statistic': ['Mean', 'Median', 'Mean', 'Median'],
    'Value': [data_attr_alloc['attr1_1'].mean(), data_attr_alloc['attr1_1'].median(), data_attr_alloc['attr4_1'].mean(), data_attr_alloc['attr4_1'].median()]
}

In [329]:
attr_stat_alloc

{'Method': ['What people look for',
  'What people look for',
  'What people think others look for',
  'What people think others look for'],
 'Statistic': ['Mean', 'Median', 'Mean', 'Median'],
 'Value': [23.95970194647202, 20.0, 31.999797242497973, 30.0]}

In [330]:
px.bar(attr_stat_alloc, x='Statistic', y='Value', color='Method', barmode='group', title='Comparison of real (blue) vs projected (red) impact of attractiveness, regardless of sex (point allocation)')

In [331]:
data_attr_alloc['gender'].value_counts()

gender
0    2467
1    2465
Name: count, dtype: int64

In [332]:
data_attr_scale['gender'].value_counts()

gender
1    781
0    776
Name: count, dtype: int64

## 3) Are shared interests more important than a shared racial background?


imprace:
How important is it to you (on a scale of 1-10) that a person you date be of the same racial/ethnic background?

Same tyopology as 
shar1_1 = 'What you look for in the opposite sex (Waves 6-9 = 1-10 scale, Waves 1-5; 10-21 = 100pt alloc)'

So we can compare both

In [333]:
shar_race = ['imprace', 'shar1_1', 'wave', 'gender']
data_imp = data[shar_race]
data_imp.head(1)

imprace  shar1_1  wave  gender
0      2.0     15.0     1       0

Some NA to drop

In [334]:
(data_imp['shar1_1'].isna().sum())/(len(data_imp))*100

1.4442587729768441

In [335]:
data_imp = data_imp.dropna(subset=['shar1_1'])

In [336]:
(data_imp['shar1_1'].isna().sum())/(len(data_imp))*100

0.0

In [337]:
(data_imp['imprace'].isna().sum())/(len(data_imp))*100

0.0

In [338]:
data_imp = data_imp.dropna(subset=['imprace'])

In [339]:
(data_imp['imprace'].isna().sum())/(len(data_imp))*100

0.0

In [340]:
len(data_imp)

8257

Ok now split by wave, especially as we demonstrated earlier how impactful it could be on results ! 

In [341]:
data_imp['wave'].value_counts().sort_index(ascending=True)

wave
1     200
2     592
3     180
4     648
5     170
6      45
7     512
8     200
9     800
10    162
11    882
12    392
13    160
14    702
15    684
16     96
17    280
18     72
19    450
20     84
21    946
Name: count, dtype: int64

In [342]:
imp_alloc = (data_imp['wave'].between(1,5)) | (data_imp['wave'].between(10,21))
imp_scale = (data_imp['wave'].between(6,9))

data_imp_alloc = data_imp[imp_alloc]
data_imp_scale = data_imp[imp_scale]

In [343]:
data_imp_scale['wave'].value_counts().sort_index(ascending=True)

wave
6     45
7    512
8    200
9    800
Name: count, dtype: int64

All good ! 

For rating approach : 

In [350]:
imp_stat_scale = {
    'Feature': ['Shared interests', 'Shared interests', 'Shared racial background', 'Shared racial background'],
    'Statistic': ['Mean', 'Median', 'Mean', 'Median'],
    'Value': [data_imp_scale['shar1_1'].mean(), data_imp_scale['shar1_1'].median(), data_imp_scale['imprace'].mean(), data_imp_scale['imprace'].median()]
}

In [347]:
px.bar(imp_stat_scale, x='Statistic', y='Value', color='Feature', barmode='group', title='Importance of Shared interests vs Shared racial background (scale 0-20)')

And now for point allocation method : 

In [349]:
imp_stat_alloc = {
    'Feature': ['Shared interests', 'Shared interests', 'Shared racial background', 'Shared racial background'],
    'Statistic': ['Mean', 'Median', 'Mean', 'Median'],
    'Value': [data_imp_alloc['shar1_1'].mean(), data_imp_alloc['shar1_1'].median(), data_imp_alloc['imprace'].mean(), data_imp_alloc['imprace'].median()]
}

In [351]:
px.bar(imp_stat_alloc, x='Statistic', y='Value', color='Feature', barmode='group', title='Importance of Shared interests vs Shared racial background (point allocation)')

Clear trend here ! Shared interests is definitively more important than shared racial background based on both sex declarations about what they are looking for in opposite sex

## Can people accurately predict their own perceived value in the dating market?

attr_o: rating by partner (pid) the night of the event, for all 6 attributes

attr3_1 = 'What is your opinion on your own (iid) attributes? (1-10 scale)'

Both are rated on a scale between 0-10, so no need to isolate waves

In [358]:
data['attr_o'].value_counts().sort_index(ascending=True)

attr_o
0.0        8
1.0      108
2.0      244
3.0      390
3.5        1
4.0      748
5.0     1260
6.0     1655
6.5        7
7.0     1642
7.5        3
8.0     1230
8.5        1
9.0      540
9.5        3
9.9        1
10.0     324
10.5       1
Name: count, dtype: int64

In [356]:
data['attr3_1'].value_counts().sort_index(ascending=True)

attr3_1
2.0       20
3.0      145
4.0      238
5.0      642
6.0     1100
7.0     2914
8.0     2217
9.0      729
10.0     268
Name: count, dtype: int64

In [362]:
# Define columns of interest
own_rate = ['attr3_1', 'sinc3_1', 'intel3_1', 'fun3_1', 'amb3_1']
partn_rate = ['attr_o', 'sinc_o', 'intel_o', 'fun_o', 'amb_o']

Note = There is no shar3_1 (Mostly omission), so I am dropping shared interests for this analysis. No big problemn as it amongst the least desirable features for both sex

In [361]:
data.head(0)

Empty DataFrame
Columns: [iid, id, gender, idg, condtn, wave, round, position, positin1, order, partner, pid, match, int_corr, samerace, age_o, race_o, pf_o_att, pf_o_sin, pf_o_int, pf_o_fun, pf_o_amb, pf_o_sha, dec_o, attr_o, sinc_o, intel_o, fun_o, amb_o, shar_o, like_o, prob_o, met_o, age, field, field_cd, undergra, mn_sat, tuition, race, imprace, imprelig, from, zipcode, income, goal, date, go_out, career, career_c, sports, tvsports, exercise, dining, museums, art, hiking, gaming, clubbing, reading, tv, theater, movies, concerts, music, shopping, yoga, exphappy, expnum, attr1_1, sinc1_1, intel1_1, fun1_1, amb1_1, shar1_1, attr4_1, sinc4_1, intel4_1, fun4_1, amb4_1, shar4_1, attr2_1, sinc2_1, intel2_1, fun2_1, amb2_1, shar2_1, attr3_1, sinc3_1, fun3_1, intel3_1, amb3_1, attr5_1, sinc5_1, intel5_1, fun5_1, amb5_1, dec, attr, sinc, ...]
Index: []

In [363]:
data_own_rate = data[own_rate]
data_partn_rate = data[partn_rate]

In [373]:
data_own_rate.rename(columns={'attr3_1': 'Attractive', 'sinc3_1': 'Sincere', 'intel3_1': 'Intelligent', 'fun3_1': 'Fun', 'amb3_1': 'Ambitious'}, inplace=True)

C:\Users\romai\AppData\Local\Temp\ipykernel_5224\3194222390.py:1: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy



In [375]:
data_partn_rate.rename(columns={'attr_o': 'Attractive', 'sinc_o': 'Sincere', 'intel_o': 'Intelligent', 'fun_o': 'Fun', 'amb_o': 'Ambitious'}, inplace=True)

C:\Users\romai\AppData\Local\Temp\ipykernel_5224\3629792970.py:1: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy



In [383]:
(data_own_rate - data_partn_rate).mean()

Attractive     0.902107
Sincere        1.126297
Intelligent    1.037478
Fun            1.309353
Ambitious      0.810194
dtype: float64

In [407]:
corr_rate = data_own_rate.corrwith(data_partn_rate)
corr_rate.sort_values(ascending=True, inplace=True)

In [409]:
corr_rate_plot = px.histogram(corr_rate, x=corr_rate.index, y=corr_rate.values, title='Correlation between predicted and actual ratings received by partners (scale 0-10)')
corr_rate_plot.update_layout(yaxis_title='Correlation strenght')
corr_rate_plot.show()

Globally, people did not predict well their attribute rating 

People predicted better their attractiveness than their sincerity, which could greatly resumes these generally poor results ! ;) 